src/dependency.cr

```diff
@@ -7,7 +7,15 @@ module Shards
     def self.new(pull : YAML::PullParser) : self
       Dependency.new(pull.read_scalar).tap do |dependency|
         pull.each_in_mapping do
-          dependency[pull.read_scalar] = pull.read_scalar
+          key = pull.read_scalar
+          value = pull.read_scalar
+
+          # HACK: user/repo is case insensitive in github repositories, we thus
+          # linearize the dependency definition to always be lowercase to avoid
+          # issues later (e.g. cloning a repository multiple times in the cache)
+          value = value.downcase if key == "github"
```

Comment:
Isn't GitLab doing the same?

Reply:
No idea.

Reply:
From my testing it seems it does.

Reply:
Yes, and bitbucket as well. I don't think any of them has case-sensitive names.
In fact, the case-sensitive resolvers are `path` (depending on the file system/os) and generic `git` URLs (partial, for example host is case-insensitive). The latter can be normalized with the improved `URL#normalize` in Crystal 0.29.0.

Reply:
Git can use whatever URL scheme, for example `file:` and I don't think the path of `file:///path/to/repo` is case insensitive.

Reply:
Yes, but in any URI scheme, the generic host component and the scheme are case-insensitive. They can and should be normalized.